bot.py

```diff
@@ -96,44 +94,50 @@ async def get_rank(ctx, name: str):
     try:
         summoner_info = get_summoner_rank(name)
 
-        embed = discord.Embed(title="Solo/Duo Rank", color=discord.Color.dark_gray())
-
-        summoner_name = summoner_info["user_name"]
-
-        # Removing space of the summoner name to access op.gg url of the summoner
-        summoner_name_opgg = summoner_name.replace(" ", "")
-
-        # Add author, thumbnail, fields, and footer to the embed
-        embed.set_author(
-            name=summoner_name,
-            url=f"https://na.op.gg/summoner/userName={summoner_name_opgg}",
-            icon_url=summoner_info["summoner_icon_image_url"],
+        embed_data = EmbedData()
+        embed_data.title = "Solo/Duo Rank"
+        embed_data.description = (
+            f"`All Data from NA server`\n\n <@!{bot.user.id}> <command>"
         )
+        embed_data.color = discord.Color.dark_gray()
 
-        # Get image of tier by path
-        tier_image = summoner_info["tier_image"]
-        file = discord.File(tier_image)
-        # Need to get the filename in order to attach to the thumbnail
+        # Add author, thumbnail, fields, and footer to the embed
+        embed_data.author = {}
+        embed_data.author = {
+            "name": summoner_info["user_name"],
+            # For op.gg link, we have to remove all whitespace.
+            "url": "https://na.op.gg/summoner/userName={0}".format(
+                summoner_info["user_name"].replace(" ", "")
+            ),
+            "icon_url": summoner_info["summoner_icon_image_url"],
+        }
+
+        # Upload tier image to discord to use it as thumbnail of embed using full path of image.
+        file = discord.File(summoner_info["tier_image_path"])
 
-        tier_image_filename = tier_image.replace("ranked-emblems/", "")
         # Embed thumbnail image of tier at the side of the embed
-        embed.set_thumbnail(url=f"attachment://{tier_image_filename}")
+        # Note: This takes the 'file name', not a full path.
+        embed_data.thumbnail = "attachment://{0[tier_image_name]}".format(summoner_info)
```

Comment:
@jung1411 This was the issue with image not showing up as thumbnail. 
We send file using the **full path** of the file, but `attachement` expects file name only. 
You were just replacing `ranked-emblems` with blank so file name was wrong.
eg; full path; `/users/stanley/discord/ranked-emblems/silver.png`
your way causes file name to be;  `/users/stanley/discord/silver.png` instead of `silver.png`